Fixed the validateBufferSize issue

zhangyichix opened the pull request:
when we use format HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED to allocate the buffer, we get the buffer with format DRM_FORMAT_XBGR8888. The validateBufferSize will return an error because format mismatch. So we should ignore this type.

Tracked-On: OAM-110999

zhangyichix commented:
@feijiang1 @renchenglei @chenyanxzhu Could you pls take a review?

feijiang1 commented on cros_gralloc/gralloc4/CrosGralloc4Mapper.cc:
```diff
@@ -149,16 +150,9 @@ Return<Error> CrosGralloc4Mapper::validateBufferSize(void* rawHandle,
-#else
-    if (descriptor.format != crosHandleFormat) {
+
+    if (descriptor.format != crosHandleFormat && !is_flex_format(cros_gralloc_convert_format((int32_t)descriptor.format))) {
```
When allocate HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED, we treat it as DRM_FORMAT_XBGR8888. Then when we validate, can we also check HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED and DRM_FORMAT_XBGR8888 are matched.

zhangyichix replied:
yes, true, this will be more rigorous.

zhangyichix commented:
hi @feijiang1 , I rewrote the code, gralloc1 was also changed, so that keep the behavior of gralloc1 and gralloc4 are same. Could you please help me to take a review again?

feijiang1 reviewed (approved):
LGTM

sysopenci reviewed (commented):
Failure when executing preintegration builder

FAILURE:CheckApprovals Some checks are failed. See the step results.html

More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/5186

sysopenci reviewed (commented):
PR integrated into 'preintegration' builder



More details, logs, and binaries: /absp/builders/celadon-preintegration/builds/5188

Done by "Zhang, YichiX" <yichix.zhang@intel.com>

sysopenci reviewed (commented):
PR integrated into 'premerge' builder



More details, logs, and binaries: /absp/builders/celadon-premerge/builds/3016

Done by "Zhang, YichiX" <yichix.zhang@intel.com>

sysopenci reviewed (commented):
cactus finished building and testing your changes 

It was requested for merge by "Zhang, YichiX" <yichix.zhang@intel.com> 

so we merge it. Thanks for your contribution.

more details, logs, and binaries: /absp/builders/celadon-mergerequest/builds/3375
